src/carousel/carousel.ts

```diff
@@ -55,24 +55,34 @@ export class NgbSlide {
     'class': 'carousel slide',
     '[style.display]': '"block"',
     'tabIndex': '0',
-    '(keydown.arrowLeft)': 'keyboard && prev(NgbSlideEventSource.ARROW_LEFT)',
-    '(keydown.arrowRight)': 'keyboard && next(NgbSlideEventSource.ARROW_RIGHT)'
+    '(keydown.arrowLeft)': 'keyboard && arrowLeft()',
+    '(keydown.arrowRight)': 'keyboard && arrowRight()',
+    '(mouseenter)': 'mouseHover = true',
+    '(mouseleave)': 'mouseHover = false',
+    '(focusin)': 'focused = true',
+    '(focusout)': 'focused = false',
+    '[attr.aria-activedescendant]': 'activeId'
   },
   template: `
-    <ol class="carousel-indicators" *ngIf="showNavigationIndicators">
+    <ol class="carousel-indicators" [class.sr-only]="!showNavigationIndicators" role="tablist">
       <li *ngFor="let slide of slides" [id]="slide.id" [class.active]="slide.id === activeId"
-          (click)="select(slide.id, NgbSlideEventSource.INDICATOR)"></li>
+          role="tab" [attr.aria-labelledby]="slide.id + '-slide'" [attr.aria-controls]="slide.id + '-slide'"
+          [attr.aria-selected]="slide.id === activeId"
+          (click)="focus();select(slide.id, NgbSlideEventSource.INDICATOR);"></li>
     </ol>
     <div class="carousel-inner">
-      <div *ngFor="let slide of slides" class="carousel-item" [class.active]="slide.id === activeId">
+      <div *ngFor="let slide of slides; let i=index;" class="carousel-item" [class.active]="slide.id === activeId" [id]="slide.id + '-slide'" role="tabpanel">
+        <span class="sr-only" i18n="Currently selected slide number read by screen reader@@ngb.carousel.slide">
+          Slide {{i + 1}} of {{slides.length}}
```

Comment:
`@@ngb.carousel.slide` sounds too generic to me here. what about `@@ngb.carousel.slide-number`